w11h03/test/pgdp/networking/SocketTest.java

```diff
@@ -283,36 +283,36 @@ void testMessages() throws IOException, InterruptedException {
-                .or(() -> assertArrayEquals(new byte[]{0x01, 0x0D}, actualLength))
-                .withMessage("Incorrect message length")
-                .run();
-
-        String actualMessage = new String(buffer, StandardCharsets.UTF_8).substring(3);
-        int messageLength = actualMessage.length();
+        if (actualLength[0] == 0x00 || actualLength[0] == 1 && actualLength[1] > 0 && actualLength[1] < 0x0D) {
+            assertTrue(actualLength[1] >= 0x0D, "Message length too small. Expected (270 <= length).");
+        }
+        if (actualLength[0] != 0x01 || actualLength[1] != 0x0D) {
+            System.err.println("[WARNING] The length bytes of your message might be incorrect.");
+            System.err.println("[WARNING] Due to the restrictions placed on me by the Übungsleitung, I cannot dynamically check whether your message size actually matches.");
+            System.err.println("[WARNING] Messages may end with a series of null-bytes. How many bytes there are, and consequently, how long your message is, is dependent on your implementation and OS.");
+            System.err.println("[WARNING] Bytes received: " + Arrays.toString(actualLength) + ", Actual length: " + actualMessageLength);
+            System.err.println("[WARNING] You will need to check if these values match by hand. Sorry :(");
+        }
 
         // THIS REMOVES NULL-BYTES AT THE END OF THE STRING AND DELIBERATELY ALLOWS FOR A FAULTY REPRESENTATION OF THE MESSAGE
         // This is because the transformation into a byte-array is done incorrectly in the template
         // https://zulip.in.tum.de/#narrow/stream/1525-PGdP-W11H03/topic/StandardCharsets.2EUTF_8.2Eencode.28message.29.2Earray.28.29/near/909726
-        if (buffer.length == 294) {
-            actualMessage = actualMessage.replace("\00", "");
-        }
-        assertEquals(message, actualMessage, "Incorrect message content.");
-
-        // Assert end of output
-        assertEquals(messageLength + 7, buffer.length, "Transferred bytes did not end after expected message.");
+        String trimmedMessage = actualMessage.substring(0, message.length());
+        assertTrue(actualMessage.substring(message.length()).matches("\\x00*"), "Message did not terminate with 0 or more null-bytes.");
+        assertEquals(message, trimmedMessage, "Incorrect message content.");
     }
 
     @Test
     @Order(9)
     @DisplayName("sendMessage() – huge")
     void testMessagesHuge() throws IOException, InterruptedException {
         // Ideally, we would use resources here, but that seems difficult with the current tests setup
-        String lipsum = Files.readString(Path.of("./test/pgdp/networking/lipsum.txt"));
-        String lipsumTruncated = Files.readString(Path.of("./test/pgdp/networking/lipsum_truncated.txt"));
+        String lipsum = Files.readString(Path.of("./test/pgdp/networking/lipsum.txt")).replace("\r", "");
```

Comment:
There are platforms using only '\r' as EOL. This change adds support for these platforms.

```suggestion
        String lipsum = Files.readString(Path.of("./test/pgdp/networking/lipsum.txt")).replaceAll("\\r\\n?", "\n");
```